tests/test_err.py

```diff
@@ -0,0 +1,28 @@
+# Testing use of cpl_errs
+
+import pytest
+
+import rasterio
+from rasterio.errors import RasterioIOError
+
+
+def test_io_error(tmpdir):
+    with pytest.raises(RasterioIOError) as exc_info:
+        rasterio.open(str(tmpdir.join('foo.tif')))
+    msg, = exc_info.value.args
+    assert msg.startswith("'{0}'".format(tmpdir.join('foo.tif')))
+    assert ("does not exist in the file system, and is not recognised as a "
```

Comment:
@sgillies with GDAL 2.1-dev, `test_io_error` is failing for me.

```
-> assert msg.startswith("'{0}'".format(tmpdir.join('foo.tif')))
(Pdb++) msg
u'No such file or directory'
(Pdb++) "'{0}'".format(tmpdir.join('foo.tif'))
"'/private/var/folders/c3/sjz4d98x1qvg040bty83_8940000gn/T/pytest-of-mperry/pytest-150/test_io_error0/foo.tif'"
```
It works when compiling against my homebrewed GDAL 1.11.3.